purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -399,8 +400,7 @@ class Purchases internal constructor(
     ) {
         startProductChange(
             activity,
-            storeProduct,
-            storeProduct.defaultOption,
+            storeProduct.defaultOption?.purchasingData ?: storeProduct.purchasingData,
```

Comment:
mayyyybe storeProduct.purchasingData's `get` could have this logic? or perhaps that's just more confusing 🤷

Reply:
Ohh, I'm torn on this! I _kind_ of like it because this isn't super clear what its doing but also... that is pretty specific for this 🤔 I don't know haha

Reply:
just thinking for future devs, how will they know which purchasingData to pass to a purchase? i assume this is kinda like the subs vs inapp verison of purchasingData, right?

Reply:
actually, it looks like this logic is duplicated in the get of `GoogleStoreProduct` anyway, so we could just use `storeProduct.purchasingData` here! of course will need to reconsider for amazon...